lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -3091,6 +3091,11 @@ class DecomposeAtenCopyOp : public OpRewritePattern<AtenCopyOp> {
       return rewriter.notifyMatchFailure(
           op, "expected result type to have a dtype");
     }
+    auto srcTy = op.getSrc().getType().cast<BaseTensorType>();
```

Comment:
Will it happen that `resultType` has DType and `srcTy` doesn't? I am curious about this because there is a check for `DType` for `resultType` just above this.
What happens if we replace `if (!resultType.hasDtype()) {` by `if (!resultType.hasDtype() or !resultType.hasSizes()) {`?